include/kframework/lemmas/bytes-simplification.k

```diff
@@ -7,19 +7,63 @@ module BYTES-SIMPLIFICATION [symbolic]
   // Buffer Reasoning
   // ########################
 
-    rule BA:Bytes       ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                    [simplification, concrete(BA)]
-    rule #buf(32, DATA) ==K BA:Bytes       =>          DATA  ==Int #asInteger(BA) requires lengthBytes(BA) <=Int 32 [simplification, concrete(BA)]
+    rule [bytes-not-equal-length]:
+      BA1:Bytes ==K BA2:Bytes => false
+      requires lengthBytes(BA1) =/=Int lengthBytes(BA2)
+      [simplification]
+
+    rule [bytes-equal-concat-split-k]:
+      A:Bytes +Bytes B:Bytes ==K C:Bytes +Bytes D:Bytes => A ==K C #And B ==K D
+      requires lengthBytes(A) ==Int lengthBytes(C)
+        orBool lengthBytes(B) ==Int lengthBytes(D)
+      [simplification]
```

Comment:
This is not correct, I believe. Here, on the RHS, we are saying set-intersection between the truth/false values of the two. So, for example, if we had `{ A ==K C #Equals true }`, and `{ B ==K D #Equals false }`, the RHS of this expression would be `#Bottom`, not `false` as intended.

I think we can just use `andBool` on the RHS here instead of `#And`?

Reply:
Oh no no no no, that has to be `andBool`, my mistake...